CMakeLists.txt

```diff
@@ -148,6 +150,13 @@ if(OC_DNS_LOOKUP_IPV6_ENABLED)
     list(APPEND PRIVATE_COMPILE_DEFINITIONS "OC_DNS_LOOKUP_IPV6")
 endif()
 
+if(OC_PUSH_ENABLED)
```

Comment:
From the `#ifdef` I deduce that `OC_PUSH` required to `OC_DYNAMIC_ALLOCATION` and `OC_COLLECTIONS_IF_CREATE`. In that case I would copy how `OC_CLOUD_ENABLED` is handled (eg. it forces the required options to be enabled):
```
if(OC_CLOUD_ENABLED)
    list(APPEND PUBLIC_COMPILE_DEFINITIONS "OC_CLOUD")
    set(OC_TCP_ENABLED ON)
    set(OC_IPV4_ENABLED ON)
endif()
```

So I would change this to:
```
if(OC_PUSH_ENABLED)
    list(APPEND PRIVATE_COMPILE_DEFINITIONS "OC_PUSH")
    if(OC_PUSHDEBUG_ENABLED)
        list(APPEND PRIVATE_COMPILE_DEFINITIONS "OC_PUSHDEBUG")
    endif()
    set(OC_DYNAMIC_ALLOCATION_ENABLED ON)
    set(OC_COLLECTIONS_IF_CREATE_ENABLED ON)
endif()
```
And these lines have to be moved before the forced variables are first used - that means before line 87 where `OC_DYNAMIC_ALLOCATION_ENABLED` is checked.

Reply:
Thanks, fixed !